File: lib/src/domain/entities/query_condition.dart
import 'package:equatable/equatable.dart';

import 'package:query_wizard/domain.dart';

class QueryCondition extends Equatable implements IEntity {
  const QueryCondition({
    required this.id,
    required this.isCustom,
    this.leftField,
    required this.logicalCompareType,
    this.rightField,
    required this.customCondition,
  });

  const QueryCondition.empty()
      : id = '',
        isCustom = false,
        leftField = null,
        logicalCompareType = LogicalCompareType.equal,
        rightField = null,
        customCondition = '';

  @override
  final String id;
  final bool isCustom;
  final QueryElement? leftField;
  final LogicalCompareType logicalCompareType;
  final QueryElement? rightField;
  final String customCondition;

  @override
  List<Object?> get props => [
        id,
        isCustom,
        leftField,
        logicalCompareType,
        rightField,
        customCondition,
      ];
}

File: server/src/QueryWizard.Domain/src/Entities/QueryCondition.cs
using QueryWizard.Domain.Enums;

namespace QueryWizard.Domain.Entities
{
    public class QueryCondition
    {
        public string Id { get; set; }

        public bool IsCustom { get; set; }

        public QueryElement LeftField { get; set; }

        public LogicalCompareType LogicalCompareType { get; set; }

        public QueryElement RightField { get; set; }

        public string CustomCondition { get; set; }
    }
}
